src/main/scala/rise/core/HighLevelConstructs.scala

```diff
@@ -51,6 +51,10 @@ object HighLevelConstructs {
     fun(a => fun(b => rec(n, a, b)))
   }
 
+  val takeBut: Expr = nFun(n =>
```

Comment:
I would suggest renaming this to `dropLast` because it effectively drops the `n` last elements in the array.